code-dot-org/code-dot-org
# Adding support for scheduled campaigns on the Congrats page

As a Korean student participating in the June/July Coding Party event, I want to see a link to the custom Minecraft Hero and AI Oceans certificates created by my government.

As a Platform Engineer, I want to be able to schedule custom certificates to exist for only a specific amount of time.

* Updates the extra links feature of our Congrats page to support date ranges.
* Adds the query string parameter `currentDate` so we can preview what the Congrats page will look like on a particular date.
* Adds Korean Coding Party campaign certificates for the time specified in the JIRA.

## Links
[Jira](https://codedotorg.atlassian.net/browse/P20-948)

## Video/Screenshots
https://github.com/code-dot-org/code-dot-org/assets/1372238/052740e8-7d58-4a1b-9e2c-f4ca633797be

Storybook
![Screenshot 2024-06-05 at 3 45 12 PM](https://github.com/code-dot-org/code-dot-org/assets/1372238/1974593b-732e-438e-91b4-8ddcaa643279)

## Testing story
Storybook

Manual testing:
* Visit http://localhost-studio.code.org:3000/s/hero/lessons/1/levels/1/lang/ko
* Click the "I'm finished" link to view the Congrats page.
* Observe there is no "extra link" under the certificate preview.
* Append `&currentDate=2024-07-20` to the URL and visit it.
* Observe there is now a link to a custom certificate under the certificate preview image.


## Deployment strategy

## Follow-up work
* Create a Jira task to delete this campaign in August
* Setup a PR to support the custom certificate for Music Lab.

## PR Checklist:
